src/esp/assets/GenericMeshData.cpp

```diff
@@ -44,18 +44,16 @@ Magnum::GL::Mesh* GenericMeshData::getMagnumGLMesh() {
   return &(renderingBuffer_->mesh);
 }
 
-void GenericMeshData::setMeshData(Magnum::Trade::AbstractImporter& importer,
-                                  int meshID) {
-  /* Guarantee mesh instance success */
-  CORRADE_INTERNAL_ASSERT_OUTPUT(meshData_ = importer.mesh(meshID));
+void GenericMeshData::setMeshData(
+    Corrade::Containers::Optional<Magnum::Trade::MeshData>& meshData) {
```

Comment:
Because the function takes ownership and expects the mesh data is valid, I would change the signature to

```suggestion
void GenericMeshData::setMeshData(Magnum::Trade::MeshData&& meshData) {
```

That (in my eyes at least) conveys better what is the function going to do with the argument.

The private `meshData_` member could still stay an optional (because the class needs an empty state I guess), but could also be just a plain `Trade::MeshData` with no index buffer, no attributes and 0 vertices by default.